osu.Game.Rulesets.Mania/UI/Column.cs

```diff
@@ -31,109 +27,52 @@ public class Column : ScrollingPlayfield, IKeyBindingHandler<ManiaAction>, IHasA
 
         public readonly Bindable<ManiaAction> Action = new Bindable<ManiaAction>();
 
-        private readonly ColumnBackground background;
-        private readonly ColumnKeyArea keyArea;
-        private readonly ColumnHitObjectArea hitObjectArea;
+        protected abstract Container HitTargetContainer { get; }
+        protected abstract Drawable KeyArea { get; }
+        protected abstract Container ExplosionContainer { get; }
+        protected internal abstract Container TopLevelContainer { get; }
 
-        internal readonly Container TopLevelContainer;
-        private readonly Container explosionContainer;
-
-        public Column(int index)
+        protected Column(int index)
         {
             Index = index;
 
             RelativeSizeAxes = Axes.Y;
             Width = COLUMN_WIDTH;
+        }
 
-            background = new ColumnBackground { RelativeSizeAxes = Axes.Both };
-
-            Container hitTargetContainer;
-
-            InternalChildren = new[]
+        public static Column CreateColumnFromType(ColumnType type, int index)
+            => type switch
```

Comment:
I don't think we have a consensus on switch expressions yet. Might be better suited as a plain old switch.